### ISSUE TITLE
Support for KISS FC

### ISSUE THREAD

**Issue by gregorg** [ShikOfTheRa/scarab-osd#180]
The Kiss flight controller from Flyduino.net has his own firmware: http://ultraesc.de/downloads/KISS_OSD.zip

Kiss FC + Kiss 24A ESC adds sensors: temperature and current conumption per ESC. Also, global current consumption in mA is very useful.

Is there any plan to add it ? Or can you tell me the way to add it myself ?

**Comment by ShikOfTheRa** [comment]
This is quite cool. Not seen before.

Unfortunately couldn't be added if using with a FC like a naze. Is that what you mean?

**Comment by gregorg** [comment]
Nope. Kiss FC is a flight-controller, with proprietary firmware. It's supports telemetry and add those sensors.

**Comment by ShikOfTheRa** [comment]
OK I wasn't aware they had an FC Assumed it came from serial bus to ESC or soemthing. Even cooler !!

Had a look at their web page. Seems it's closed source so no info. Had a quick look at the OSD code and it is not using MSP so would have to reverse eng.

As a result - too much time so sorry no plans from me... 
Would be happy for someone to add in support subject to being able to exclude additional code via #defines. I think it's not to difficult...

**Comment by ShikOfTheRa** [comment]
gregorg - if the kiss guys would like to add this support into MWOSD can you get them to contact me.

**Comment by gregorg** [comment]
I'm in contact with Felix from Flyduino, he will transmit me the kiss serial protocol.

Please re-open this issue ;)

**Comment by gregorg** [comment]
the serial connection is 8bit 115200baud full duplex you can get the datas at serial1 (RX/TX pins next to the USB port) the FC will only answer to requests of on the serial, as long as there is no USB plugged in.

to request the telemetry data simply write 0x20 to it. the replay function lookes like this:

	// GUI viewable variables
	
	serial_send_byte(5); // startByte
	serial_send_byte(154); // BytesNum the byte count that will be sent
	
	serial_send_floats(RXcommands,8); // 16
	serial_send_byte(Armed); // 17
	serial_send_word((uint16_t)(LiPoVolt*100)); // 19
	serial_send_words(GyroXYZ,3); //25
	serial_send_words(ACCXYZ,3); // 31

	serial_send_word((int16_t)(angle[0]*100)); // 33
	serial_send_word((int16_t)(angle[1]*100)); // 35
	serial_send_word((int16_t)(angle[2]*100)); // 37
	
	serial_send_word(I2C_Errors); // 39
	serial_send_word(calibGyroDone); //41
	serial_send_byte(failsave); //42
	serial_send_word((uint16_t)debug*1000); //44
	serial_send_byte(0); //45
	
	serial_send_floats(GyroRaw,3); //51
	serial_send_floats(ACCRaw,3); //57
	
	serial_send_floats(ACCtrim,2); // 61
	serial_send_floats(ACCAng,2); // 65
	serial_send_byte(mode); // 66
	serial_send_word((uint16_t)debug1*1000); //68
	serial_send_words(PWMOutVals,6); // 80
	serial_send_word((uint16_t)debug2*1000); //82
	serial_send_byte(idleTime);//83
	
	serial_send_words(ESC_Telemetrie[0],5); // 93
	serial_send_words(ESC_Telemetrie[1],5); // 103
	serial_send_words(ESC_Telemetrie[2],5); // 113
	serial_send_words(ESC_Telemetrie[3],5); // 123
	serial_send_words(ESC_Telemetrie[4],5); // 133
	serial_send_words(ESC_Telemetrie[5],5); // 143
	serial_send_byte(MaxTemp); // 144
	serial_send_word(MinVolt); // 146
	serial_send_word(MaxAmps); // 148
	serial_send_word(usedMAH); // 150
	serial_send_word(MaxRPMs); // 152
	serial_send_word(MaxWatt); // 154
	
	
	serial_send_byte(checkSum); // endByte 

- the channel order is throttle, roll, pitch, yaw, aux 1234

- gyro and acc datas are send X,Y,Z//0,1,2

- the ESC's telemetry is for each ESC 

  0: temperature in 1°C, 
  1: voltage in 0.01V 
  2: current in 0.01A
  3: consumption in 1mah
  4: erpm in 100erpm

- the checksum becomes useless as there are too many bytes now. i will add a real CRC in future. if you still want to use it, it is the average of all bytes send (excluding the two start bytes and its self)

- floats are used *1000 so (float)1.0f == (int16_t)1000

- 16bit wariables are sent with high byte first

**Comment by gregorg** [comment]
Can you re-open it ?

**Comment by ShikOfTheRa** [comment]
If anyone picks this up... take not of 
  #define SERIALBUFFERSIZE 150 in serial.ino

It will need to be amended to suit - but also note will have to verify stack/heap memory is not blown

**Comment by matolowa** [comment]
hi ,gregorg
If is  possible please write to kiss OSD stopwatch .i use it in reduce mode.Bottom middle would be nice. :) thank you

**Comment by gregorg** [comment]
@matolowa you mean on original Kiss OSD firmware ?

**Comment by matolowa** [comment]
yes,or is it impossible?

**Comment by gregorg** [comment]
it is, but it's not the purpose of this issue ;) I've created a github repository to enhance the original Kiss OSD here: https://github.com/gregorg/KISS_OSD
Please create a new issue on this repository.

**Comment by ShikOfTheRa** [comment]
@gregorg - If  prepare the MWOSD to theoretically work with KISS OSD would you be interested in testing and fine tuning any variable mapping. 
I'll get it to the point where its probably 90%

**Comment by matolowa** [comment]
greg i´m interested to testing you work.your kiss osd everyone used in
Slovakia :),,everyone is satisfied ..nickname is the best :) If we fly
five..

2016-04-11 12:56 GMT+02:00 ShikOfTheRa <notifications@github.com>:

**Comment by gregorg** [comment]
@ShikOfTheRa I'm sorry I forgot to reply to your proposal ... Sure, I'm OK to test it ! I'm glade to see that it is ready for testing !
I'll have a fresh kiss fc and micro mini osd to test it soon.

**Comment by ShikOfTheRa** [comment]
Its not ready yet! Just completed mavlink ready for testing. Probably LTM next then KISS. I'll notify you when have something. Thanks for reply :)

**Comment by matolowa** [comment]
ok.I will be waiting..

2016-05-03 9:17 GMT+02:00 ShikOfTheRa <notifications@github.com>:

**Comment by ShikOfTheRa** [comment]
Can anyone clarify:
For RC channels:
Guide above indicates RC data is sent as float..., but values are 16 bit rather than 32bit float so is this half float (unusual)
serial_send_floats(RXcommands,8); // 16

In the code, the data appears to contradict and indicate its 16 bit it with a range of prob -500 to +500??

I hope its the latter :)

**Comment by gregorg** [comment]
If it helps you, take a look at the original Kiss OSD source code: https://github.com/gregorg/KISS_OSD/blob/master/src/KISS_OSD.ino

**Comment by ShikOfTheRa** [comment]
Yeah in that code it appears that it is 16 bit int. with range of at least -250 to +250

Ah well I've done it like that and ignore the float method. Lets see what happens when someone tests!

**Comment by ShikOfTheRa** [comment]
Also I think mode just determines full screen/minimal screen. 
Are there any different flight modes in KISS. Sorry for dumbass questions - just never seen / used !

Think I'll just do the basics and let someone else with kit to test  develop further.

**Comment by gregorg** [comment]
there is 2 flight modes in Kiss: acro and horizon

**Comment by ShikOfTheRa** [comment]
In the code, the mode indicator seems to change the screen displayed rather than indicate acro or horizon. Cant see anything for flight mode.

Welcome to try first cut.... I have to do the amperage as its currently using method used by mavlink...
Its not easy without knowledge or testing! Probably find anumber of items not quite correct!
https://github.com/ShikOfTheRa/scarab-osd/tree/SHIKI

Once get basics done, can add in individual ESC info etc.

**Comment by llambkin** [comment]
Id love to test, but I have no coding skills, so I dont know if I would be of any help....

**Comment by llambkin** [comment]
OK....I have thrown the program on the OSD, The voltage appears to work ok, as does the amperage counter. However, when I have finished a flight it shows the statistics for a split second only.

Quick question. Do you think that KISS would allow for the changing of PIDs in the OSD? Probably asking a bit much for a closed source FC.

**Comment by fedorcomander** [comment]
Its possible to change pids. No problem here. Protocol is rather simple and you can see it in the GUI projects on the github :) But u cannot save it while LIPO is connected (yet)...  And you need something to power vtx and osd.... 

And one more thing. float data being send as integer x1000 :)))

**Comment by ShikOfTheRa** [comment]
@fedorcomander thanks for float info. I understand now. No float being sent....
It was confusing me because it was saying float....  !!

@llambkin - thanks to fedor post above I think the throttle info would be to high making the end of flight flash off/on 

I think ...
throttle channel  is being send as 1000 + (0-1000)
all other 1500 + (-500 to +500)

Also I changed the Amperage calculation to be direct from ESC. Be interested to hear if this still works and is accurate.

Also, can you see if access the OSD menu?

test code here...
https://github.com/ShikOfTheRa/scarab-osd/tree/SHIKI

**Comment by fedorcomander** [comment]
obj.RXcommands[0] = 1000 + ((data.getInt16(i + 0, 0) / 1000) * 1000);
			obj.RXcommands[1] = 1500 + ((data.getInt16(i + 2, 0) / 1000) * 500);
			obj.RXcommands[2] = 1500 + ((data.getInt16(i + 4, 0) / 1000) * 500);
			obj.RXcommands[3] = 1500 + ((data.getInt16(i + 6, 0) / 1000) * 500);
			obj.RXcommands[4] = 1500 + ((data.getInt16(i + 8, 0) / 1000) * 500);
			obj.RXcommands[5] = 1500 + ((data.getInt16(i + 10, 0) / 1000) * 500);
			obj.RXcommands[6] = 1500 + ((data.getInt16(i + 12, 0) / 1000) * 500);
			obj.RXcommands[7] = 1500 + ((data.getInt16(i + 14, 0) / 1000) * 500);
			obj.Armed = data.getUint8(i + 16);
			obj.LiPoVolt = data.getInt16(i + 17, 0) / 1000;
			obj.GyroXYZ[0] = data.getInt16(i + 19, 0);
			obj.GyroXYZ[1] = data.getInt16(i + 21, 0);
			obj.GyroXYZ[2] = data.getInt16(i + 23, 0);
			obj.ACCXYZ[0] = data.getInt16(i + 25, 0);
			obj.ACCXYZ[1] = data.getInt16(i + 27, 0);
			obj.ACCXYZ[2] = data.getInt16(i + 29, 0);
			obj.angle[0] = data.getInt16(i + 31, 0) / 1000;
			obj.angle[1] = data.getInt16(i + 33, 0) / 1000;
			obj.angle[2] = data.getInt16(i + 35, 0) / 1000;
			obj.I2C_Errors = data.getInt16(i + 37, 0);
			obj.calibGyroDone = data.getInt16(i + 39, 0);
			obj.failsave = data.getUint8(i + 41);
			obj.debug = data.getUint16(i + 42, 0) / 1000;
			obj.foundRX = data.getUint8(i + 44);

			obj.GyroRaw[0] = data.getInt16(i + 45, 0) / 1000;
			obj.GyroRaw[1] = data.getInt16(i + 47, 0) / 1000;
			obj.GyroRaw[2] = data.getInt16(i + 49, 0) / 1000;
			obj.ACCRaw[0] = data.getInt16(i + 51, 0) / 1000;
			obj.ACCRaw[1] = data.getInt16(i + 53, 0) / 1000;
			obj.ACCRaw[2] = data.getInt16(i + 55, 0) / 1000;
			obj.ACCtrim[0] = data.getInt16(i + 57, 0) / 1000;
			obj.ACCtrim[1] = data.getInt16(i + 59, 0) / 1000;
			obj.ACCAng[0] = data.getInt16(i + 61, 0) / 1000;
			obj.ACCAng[1] = data.getInt16(i + 63, 0) / 1000;
			obj.mode = data.getUint8(i + 65);
			obj.debug = data.getUint16(i + 66, 0) / 1000;
			obj.PWMOutVals[0] = data.getInt16(i + 68, 0);
			obj.PWMOutVals[1] = data.getInt16(i + 70, 0);
			obj.PWMOutVals[2] = data.getInt16(i + 72, 0);
			obj.PWMOutVals[3] = data.getInt16(i + 74, 0);
			obj.PWMOutVals[4] = data.getInt16(i + 76, 0);
			obj.PWMOutVals[5] = data.getInt16(i + 78, 0);
			obj.debug2 = data.getUint16(i + 80, 0) / 1000;
			obj.idleTime = data.getUint8(i + 82);

			obj.ESC_Telemetrie0[0] = data.getInt16(i + 83, 0);
			obj.ESC_Telemetrie0[1] = data.getInt16(i + 85, 0);
			obj.ESC_Telemetrie0[2] = data.getInt16(i + 87, 0);
			obj.ESC_Telemetrie0[3] = data.getInt16(i + 89, 0);
			obj.ESC_Telemetrie0[4] = data.getInt16(i + 91, 0);

			obj.ESC_Telemetrie1[0] = data.getInt16(i + 93, 0);
			obj.ESC_Telemetrie1[1] = data.getInt16(i + 95, 0);
			obj.ESC_Telemetrie1[2] = data.getInt16(i + 97, 0);
			obj.ESC_Telemetrie1[3] = data.getInt16(i + 99, 0);
			obj.ESC_Telemetrie1[4] = data.getInt16(i + 101, 0);

			obj.ESC_Telemetrie2[0] = data.getInt16(i + 103, 0);
			obj.ESC_Telemetrie2[1] = data.getInt16(i + 105, 0);
			obj.ESC_Telemetrie2[2] = data.getInt16(i + 107, 0);
			obj.ESC_Telemetrie2[3] = data.getInt16(i + 109, 0);
			obj.ESC_Telemetrie2[4] = data.getInt16(i + 111, 0);

			obj.ESC_Telemetrie3[0] = data.getInt16(i + 113, 0);
			obj.ESC_Telemetrie3[1] = data.getInt16(i + 115, 0);
			obj.ESC_Telemetrie3[2] = data.getInt16(i + 117, 0);
			obj.ESC_Telemetrie3[3] = data.getInt16(i + 119, 0);
			obj.ESC_Telemetrie3[4] = data.getInt16(i + 121, 0);

			obj.ESC_Telemetrie4[0] = data.getInt16(i + 123, 0);
			obj.ESC_Telemetrie4[1] = data.getInt16(i + 125, 0);
			obj.ESC_Telemetrie4[2] = data.getInt16(i + 127, 0);
			obj.ESC_Telemetrie4[3] = data.getInt16(i + 129, 0);
			obj.ESC_Telemetrie4[4] = data.getInt16(i + 131, 0);

			obj.ESC_Telemetrie5[0] = data.getInt16(i + 133, 0);
			obj.ESC_Telemetrie5[1] = data.getInt16(i + 135, 0);
			obj.ESC_Telemetrie5[2] = data.getInt16(i + 137, 0);
			obj.ESC_Telemetrie5[3] = data.getInt16(i + 139, 0);
			obj.ESC_Telemetrie5[4] = data.getInt16(i + 141, 0);

			obj.ESC_TelemetrieStats[0] = data.getInt16(i + 142, 0);
			obj.ESC_TelemetrieStats[1] = data.getInt16(i + 144, 0);
			obj.ESC_TelemetrieStats[2] = data.getInt16(i + 146, 0);
			obj.ESC_TelemetrieStats[3] = data.getInt16(i + 148, 0);
			obj.ESC_TelemetrieStats[4] = data.getInt16(i + 150, 0);
			obj.ESC_TelemetrieStats[5] = data.getInt16(i + 152, 0);

**Comment by fedorcomander** [comment]
Hope this help :)

**Comment by fedorcomander** [comment]
all this code available in https://github.com/fedorcomander/kissfc-chrome-gui

**Comment by ShikOfTheRa** [comment]
Yes - much thanks. I just saw your link above. Very helpfull

**Comment by llambkin** [comment]
Hello Guys,

I have done an actual flight with the test code from a few days ago. You are right Shik, it was flashing on/off a lot thoughout the flight. Also, yes I can access the menu, it is far more responsive than Naze F3 targets, that was maybe because the PWM was maxed a lot sooner.

I will throw the new code on there tonight and give it a test, I will report back here.

Here is a video of the flight in question.

**Comment by llambkin** [comment]
++ Thanks for the help fedorcomander. Lets hope that they fix the save feature so it allows for changing PIDs via OSD. Or is it you that is working on this?

**Comment by llambkin** [comment]
Ok....Ive tested the current build Shik. The amperage appears to work now. But the amperage counter doesnt work at all. It sat on 0 the whole time.

Also, I had a reset during flight. It said disarmed, then armed and continued as normal.

Flight video to follow.

**Comment by ShikOfTheRa** [comment]
Thanks for vid - VERY usefull.
MaH fixed - my error.
I think I know cause of the armed reset. Probably accounts for other glithes - including the High max amps on summary.

So throttle and summary screen looks OK - does the menu work OK - get into it OK, change setting and exit?  

Try to get last bits done tomorrow

**Comment by gregorg** [comment]
Awesome ! I do not have a testing frame today, I will have one in couple weeks and will test it for sure :)

Another usefull features to add is ESC temperatures, and maybe caliGyroDone ?

**Comment by mart1n** [comment]
Oh man! I'm excited to see that support for the KISS FC is being added to MWOSD. I've been having issues with the KISS OSD firmware. The entire video feed from my vtx is full of random characters, mostly the letter 'V', from top to bottom. I never loaded the KISS fonts file though, so that might be it. But I'm not going to worry about trying to get it to work and I'm going to move straight over to this branch of MWOSD. I'll probably even help with some development.

**Comment by llambkin** [comment]
Yes Shik, I am able to enter the menu, and change settings and exit :smile: 

Thanks heaps for your coding ninja skills. Will test the new code as soon as possible. I am guessing it would take a fair bit of work (changing the configurator) to add the code for individual ESC amps/temps/RPM?

**Comment by ShikOfTheRa** [comment]
Its now merged into the Master as its getting closer to being usable.. had a couple of issues with sync so hopefully OK

https://github.com/ShikOfTheRa/scarab-osd/


what needs testing (I think!)
Does it still work (have added the checksum to validate data coming through - if wrong will see nothing !)
Amps - is it accurate and smoother
Horizon bar - does it move at all / roughly track horizon / ?
Has in flight screen quick disarm glitching stopped.

@llambkin thanks. Less ninja, more turtle.... 
I want to get the basics in that MWOSD supports today. Someone can then look to add

@mart1n  - welcome on board. Happy to take any improvements

@gregorg  - wanted to get the basics in place so others can expand. It should be easy to implement though I'll probably not be adding those bits myself (not exactly easy when not having a frame with kiss controller to test! - but happy to take any submissions)

**Comment by llambkin** [comment]
@ShikOfTheRa, I will test all of the above. I don't really fly with the horizon on, but will test for those who do.

Will report back tomorrow with another video.

**Comment by mart1n** [comment]
@ShikOfTheRa - In my previous comment I mentioned that the KISS OSD firmware filled my vtx video feed with characters from top to bottom. After flashing MWOSD, the same issue exists!

If you look at the top of the KISS_OSD.ino file in the KISS OSD FW you notice the second group of macros reads...

`// MAX7456 Charset (change if you get sensless signs)
//=============================
#define USE_MAX7456_ASCII
//#define USE_MAX7456_MAXIM
`

I had to change this from USE_MAX7456_ASCII to USE_MAX7456_MAXIM. In the KISS OSD FW to fix my issue. The USE_MAX7456_* macros are used to determine the character encoding used.

I noticed the MAX7456 APIs that the KISS OSD FW uses is different from what MWOSD is using.

Does MWOSD already have a method for switching character encoding? I briefly looked for a way to switch MWOSD from MAX7456_ASCII to MAX7456_MAXIM and didn't see anything.

I will mention in my search through rcgroups.com I have found 2 other people that are having this exact issue with MWOSD firmware. It's not an issue with font loading either as I tried reloading fonts with USB power as well as with Lipo power.

Thanks!

Martin

**Comment by mart1n** [comment]
Here is an example of what my vtx video feed looks like....
[http://static.rcgroups.net/forums/attachments/5/7/6/3/3/6/a8858368-194-osderror.jpg](http://static.rcgroups.net/forums/attachments/5/7/6/3/3/6/a8858368-194-osderror.jpg)

**Comment by llambkin** [comment]
OK, I have tested the code.

It appears that the amps is working and correct.
The horizon appears to work, but is back to front.
I didnt have a long enough flight to test if the in flight glitching has stopped.

I will leave the code on there and take it for a proper test later.

Regards -Liam

**Comment by llambkin** [comment]
I have done a longer flight today.
The in flight resets are still happening, I was looking for how many amps I have used, and it only said 70mah..... Then I noticed it reset to 0, and it did this several times through the flight.

Sorry I dont have a video of this happening. I will get one tonight when I go out for an event.

**Comment by llambkin** [comment]
Hahahahaha, I just realised, the horizon isnt back to front.
So, that just leaves the reset glitch.

**Comment by ShikOfTheRa** [comment]
OK I thought adding checksum for the data would remove the "reset"  or at least minimise it.
What is interesting - the mah counter comes from the FC, so if that is resetting when you get the OSD glitch, then its likely to be that your FC is resetting that info.... Look forward to next vid. Sorry for delay - been away for  few days

**Comment by llambkin** [comment]
I will make a video tonight, I cant get it to do it in the garage.....

**Comment by llambkin** [comment]
Video as promised.....A little late, sorry :)
https://youtu.be/9AiNuXj47Rs

**Comment by ShikOfTheRa** [comment]
No worry... Perfect! Actually I think the resets are solved/improved - I didn't see any... 
The mah counter is a separate issue. I can see the problem in my code. I'll simulate test and advise back...
The frequent low vlts are annoying - i will try to put some code to improve that in the future. For now can comment this part out...
//#define DISP_LOW_VOLTS_WARNING      // Alerts with additional text warning if low voltage

**Comment by ShikOfTheRa** [comment]
OK, that done tested and uploaded.
I think only time will tell now to see if we get any more screen resets.
Thanks for your time to do the testing and uploads. I know its a pain in the butt....

I think we have the basic in place for KISS MW-OSD :) something we can build and add to.
Keep us posted if see any more issues.

**Comment by llambkin** [comment]
Awesome. I will test and report back. Its great to collaborate with nice people :D
Yeah, I have some saggy LiPo's and KISS FC spits out the data so quickly that both my TX (Taranis) and OSD bitch about voltage all the time. I think its about time to get some better LiPo's.

 Some code to put a delay of about 5 secs of low voltage would be nice. It rarely gets hammered for more than 5 seconds at a time. I have put a logical switch into my TX that triggers an announcement only after 5 seconds of sag below 10.5v

Thanks again for your skills. I am more than happy to test the code out since it is great benefit to myself and others once we get it right. Is it a big job to code the individual ESC information into the GUI and OSD, is that something that someone else is likely to have to do?

**Comment by llambkin** [comment]
I have tested the current build with my copter. It all works as expected, no resets, and I can now rely on it for mAh used :D

**Comment by llambkin** [comment]
https://youtu.be/nPt2Xlb3CXg

**Comment by ShikOfTheRa** [comment]
Awesome!! Thanks to you guys testing and nagging me I think we have at least the essentials working now. 
For adding indivdual ESC data its not difficult. Just some time. Figuring out best way to display really.

How usefull is that ESC info? Think I would prefer to have PIDS adjustable in menu over that

**Comment by llambkin** [comment]
All I would want is Temp and Amps of ESC. The current way of displaying the info is to put the info in each corner according to the layout of the motor order.

Yeah, adjusting PID via OSD would be great, but I don't know when Flyduino plan on making that available to us. They currently have a problem where if you send a save command to the FC, it flips out. So their fix for that is not being able to save with a lipo connected.

So i figured that someone could work on the features available to us right now.

**Comment by ShikOfTheRa** [comment]
OK. Ill have a think. I'd hate to fly with that info on all the time - so prob make it one of the switchable layout options

**Comment by ShikOfTheRa** [comment]
@mart1n 

Maybe the dev latest version on MWOSD will be of use to you no llabkin confirmed its looking OK. 

Alternatively, in mwosd, can flash fonts without the gui using this method:


/*--------------------------       INITIALISATION options       ----------------------------------------------------*/
// Ignore this section unless you know you need to use it !!
// This section contains initialisation options that only require to be run once.
// Once the initialisation has completed, all sections should be commented and the sketch re-uploaded.
// Font upload will take 90 seconds after upload is completed. If connected to a camera, you will see teh font table displayed.

//#define LOADFONT_DEFAULT         // Uncomment to force an upload of default font instead of using GUI

**Comment by llambkin** [comment]
I put the info to the back of my mind, I totally ignore it.I flew an entire pack without looking at the amperage today. I mainly want it for troubleshooting, its to look back on it after a flight, if one motor is a bit stuffed, then the ESC will draw substantially more current, and the ESC will get hotter.

But you are right, PIDs would be more important. Maybe you could start on just getting them to display, that would serve the purpose of having video evidence of your PIDs before you fly. Also we could do some testing on adjusting PIDs, since I can plug in a USB power pack to the FC, since I can save values when the FC is powered off USB. Also the FC powers the RX too.....but not the VTX.....hmmmmm

**Comment by ShikOfTheRa** [comment]
Hopefully have some KISS stuff on the way soon to build a test rig. Once get that can have a look at the extras unless someone wants to jump in and do them.

**Comment by llambkin** [comment]
I will always be available to test :D

**Comment by fedorcomander** [comment]
sending command is not a problem. fc will not accept it if connected to the lipo. for greater good ;)

---
Cheers,
Alex

**Comment by llambkin** [comment]
Obviously not for the greater good. The other open source FC's out there have no problem saving settings when the lipo is connected. I have no idea about coding, but it doesn't seem like a big problem for a board not to go haywire when the save button is pressed when a battery is connected.

I know that in the early days of cleanflight, there were a few close calls when people were checking the oneshot125 box with the battery connected, and betaflight still does that to this day.

Is it being worked on? or should we just forget about having such an awesome feature? Being able to tune my copter without having to bring the laptop would be a KISS thing to have.

**Comment by redwolf3773** [comment]
I loaded the new firmware to both my Kiss setup quads and so far it is working like a charm. I am only using Voltage, Timer, Throttle % so far. I am not running Kiss ESCs, I am running the V-Good Firefly's on one and ZTW HV 20amp ESCs on the other. The only issue so far has been with the GUI. It will stall at 60% sometimes or 72% and I have to disconnect and reconnect and try again. I found if I had less items selected it would make it all the way through the update. If I had a lot of items selected it would jam up. I also want to thank llambkin for steering me over here from his YouTube Channel. I have been keeping my eye out for Shik to start on the Kiss adaptation but didn't know it until I stumbled across llambkin's test videos.

**Comment by llambkin** [comment]
Hey Redwolf, I actually had this the other day, and what fixed it for me was doing an EPROM clear first before sending mwosd.
Let me know if it fixes it for you.

**Comment by ShikOfTheRa** [comment]
Yes - there are some issues if using mix of GUI / OSD versions. EEPROM clear or reset to defaults may clear. Also at one point there was a dodgy layouts file. If no joy, try  using latest released package I created today. Unfortunately no MAC in there at the moment.

**Comment by skaman82** [comment]
@ShikOfTheRa Can't use it without Mac support :(

**Comment by ShikOfTheRa** [comment]
Donations of a MAC would be welcomed :)

**Comment by skaman82** [comment]
@ShikOfTheRa I would donate one if I had the money :) How do you managed to release the mac GUI before?

**Comment by ShikOfTheRa** [comment]
lol. I wouldn't !
I had a virtual machine with hackintosh running. Unfortunately I have changed PC's and it has a smaller disk no space for virtual machine. waiting for an upgrade...

**Comment by ShikOfTheRa** [comment]
@llambkin - the voltage fluctuation on your vid bugged me. I recall you said batts were crap however I have enabled filtering for FC voltage data.

Should notice a difference straight away. If not enough, use the hysterisis filter.
Options are:
#define FILTER_AVG
//#define FILTER_HYSTERYSIS 

Also as FYI, 3 modes added for KISS - should now display STAB or 3D if enabled and mode active.

**Comment by llambkin** [comment]
That is cool, I will throw the code on right away. I will post a new video after my meet tomorrow night.

Thanks :D

**Comment by llambkin** [comment]
It is broken completely.....I have no OSD at all, until I flashed back to the previous build...

**Comment by ShikOfTheRa** [comment]
OK, that may be a different issue. PAL/NTSC setting not matching your cam. Update would have changed that. Will probably need to use the new GUI to configure settings like use FC for amps and volts.

Try enabling AUTOCAM in config.h

Unfortunately I can't test anything until next week,

**Comment by llambkin** [comment]
All good, I will throw it back on and do some more testing....

**Comment by ShikOfTheRa** [comment]
Guys - will close this as not planning any further updates until the KISS guys implement a solution for live updates of PIDS etc.   If enough resource is available once that is done I'll see if we can add in the additional ESC info

**Comment by nathantsoi** [comment]
quick fyi: betaflight will support kiss soon. then the bf commands will
apply. or do people really want to use the kiss firmware?

**Comment by ShikOfTheRa** [comment]
Thanks for the info Nathan. Always good to have options. 
Kiss has a huge following in EU. They are a quiet bunch on the forums though. Possibly because it is simple to configure and tune. 
If you check youtube for "KISS FC" you'll get a lot of hits on vids. Some look as impressive as ßF/CF

I now have one and just waiting for rest of kit to fly.

**Comment by fedorcomander** [comment]
"Some look as impressive as ßF/CF" - good one :-))))))

**Comment by ShikOfTheRa** [comment]
:) The ones that are tuned well look really good. I was really impressed with how well they performed. It made me want one ... :)

**Comment by discoltk** [comment]
Just wanted to thank you guys for getting KISS support added.   Built my first 250 quad, the TBS OSD I was using at first was just too big and I discovered the minim.  Was disappointed by the lack of configure-ability of the KISS OSD and was very happy to find out you'd already added it to the -current tree.   Count me in for any testing.   

If you'll add bitcoin as an option to your donations page I'll gladly send you a bit for your trouble.

Thanks for your hard work!

**Comment by ShikOfTheRa** [comment]
@discoltk - sorry missed this . Thanks for feedback. Haven't got bitcoin - but thanks for the offer. Thats good enough for me!

I'm going to close this down for now. When teh KISS frmware has support for live PID update please resurrect and we'll get it added.

**Comment by ShikOfTheRa** [comment]
@fedorcomander 
one of my users has communicated that PIDs can now be updated - can you share details of protocol so I can implement?

Also any changes to existing read protocol? Is it still same.

TIA

**Comment by fedorcomander** [comment]
Please check LUA sources, its all in there if you want to change ONLY pids.

**Comment by Markuzzzz** [comment]
Hey Guys, I am testing the MW Osd for Kiss. I have a strange error going on. When ever I use level/acro mode selection using a switch on my TX the osd disappears entirely. Not matter on which Aux I set level/acro mode in the Kiss Gui. Anybody having the same problem? Shouldnt it react only on AUX4?

Tried 1.6.8 and the newest beta of 1.7 for Kiss from github.
